source/server/connection_handler_impl.cc

```diff
@@ -107,6 +109,42 @@ void ConnectionHandlerImpl::enableListeners() {
   }
 }
 
+void ConnectionHandlerImpl::retryAllConnections(
+    const envoy::config::listener::v3::FilterChain* const& filter_chain_message) {
+  ENVOY_LOG(debug, "inside worker.handler.retryALLconnections");
+  if (sockets_using_filter_chain_.find(filter_chain_message) == sockets_using_filter_chain_.end()) {
+    return;
+  }
+  auto& listener_sockets_map = sockets_using_filter_chain_[filter_chain_message];
+  // Go through all listeners on this worker, if a listener has sockets sending to rebuild
+  // filter_chain, now retry connection with those sockets.
+  for (auto& listener : listeners_) {
+    if (listener.second.tcp_listener_.has_value()) {
+      ENVOY_LOG(debug, "listen has tcp");
+      auto& active_tcp_listener = listener.second.tcp_listener_->get();
+      const auto& listener_name = active_tcp_listener.config_->name();
+      ENVOY_LOG(debug, "active_tcp_listener: name: {}", listener_name);
+      // Find all SocketMetadataPair of this active tcp listener.
+      if (listener_sockets_map.find(listener_name) != listener_sockets_map.end()) {
+        ENVOY_LOG(debug, "we have saved this listener for retry connection");
+        for (auto& socket_metadata_pair : listener_sockets_map[listener_name]) {
+          // Retry connection with that socket on this active tcp listener.
+          // Should point real filter chain ahead of retry connection, to avoid meet fake filter
+          // chain again.
+          ENVOY_LOG(debug, "active_tcp_listener: newConnection");
+          active_tcp_listener.newConnection(std::move(socket_metadata_pair.first),
+                                            socket_metadata_pair.second);
+        }
```

Comment:
3. You may need to move the unique_ptr back to the list if retry fails

Reply:
Yes, and I will add a flag parameter to this retry connection function to indicate whether rebuilding is successful.